Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -0,0 +1,1301 @@
+        {
+            bindPreference = string.Empty;
+            port = null;
+
+            // Check for user override.
+            foreach (var addr in _publishedServerUrls)
+            {
+                // Remaining. Match anything.
+                if (addr.Key.Equals(IPAddress.Broadcast))
+                {
+                    bindPreference = addr.Value;
+                    break;
+                }
+                else if ((addr.Key.Equals(IPAddress.Any) || addr.Key.Equals(IPAddress.IPv6Any)) && isExternal)
+                {
+                    // External.
+                    bindPreference = addr.Value;
+                    break;
+                }
+                else if (addr.Key.Contains(source))
+                {
+                    // Match ip address.
+                    bindPreference = addr.Value;
+                    break;
+                }
+            }
+
+            if (!string.IsNullOrEmpty(bindPreference))
+            {
+                // Has it got a port defined?
+                var parts = bindPreference.Split(':');
+                if (parts.Length > 1)
+                {
+                    if (int.TryParse(parts[1], out int p))
+                    {
+                        bindPreference = parts[0];
+                        port = p;
+                    }
+                }
+
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Attempts to match the source against a user defined bind interface.
+        /// </summary>
+        /// <param name="source">IP source address to use.</param>
+        /// <param name="isExternal">True if the source is in the external subnet.</param>
+        /// <param name="result">The result, if a match is found.</param>
+        /// <returns><c>true</c> if a match is found, <c>false</c> otherwise.</returns>
+        private bool MatchesBindInterface(IPObject source, bool isExternal, out string result)
+        {
+            result = string.Empty;
+            var nc = _bindAddresses.Exclude(_bindExclusions);
+
+            int count = nc.Count;
+            if (count == 1 && (_bindAddresses[0].Equals(IPAddress.Any) || _bindAddresses[0].Equals(IPAddress.IPv6Any)))
+            {
+                // Ignore IPAny addresses.
+                count = 0;
+            }
+
+            if (count != 0)
+            {
+                // Check to see if any of the bind interfaces are in the same subnet.
+
+                Collection<IPObject> bindResult;
+                IPAddress? defaultGateway = null;
+                IPAddress? bindAddress;
+
+                if (isExternal)
+                {
+                    // Find all external bind addresses. Store the default gateway, but check to see if there is a better match first.
+                    bindResult = CreateCollection(nc
```

Comment:
Is it strictly required to create a collection just for this? It looks like a simple for loop over `nc` would do the same thing

Reply:
rewritten